[DOCS-1166][DOCS-880][DOCS-1167][DOCS-1168] Update automation docs (#1084)
wandb/docs: content/guides/core/automations/_index.md

title: Automations
weight: 4
---
{{% pageinfo color="info" %}}
{{< readfile file="/_includes/enterprise-only.md" >}}
{{% /pageinfo %}}

> This page describes _automations_ in W&B. [Create an automation]({{< relref "create-automations/" >}}) to trigger workflow steps, such as automated model testing and deployment, based on an event in W&B, such as when an [artifact]({{< relref "/guides/core/artifacts" >}}) artifact version is created ({{< relref "/guides/core/registry/" >}}).

Reviewer: Link incorrectly formatted.

Author: Done

> This page describes _automations_ in W&B. [Create an automation]({{< relref "create-automations/" >}}) to trigger workflow steps, such as automated model testing and deployment, based on an event in W&B, such as when an [artifact]({{< relref "/guides/core/artifacts" >}}) artifact version is created ({{< relref "/guides/core/registry/" >}}).

Reviewer: Nit: Add a definition after first mention of automations, following the italicization of the word.

Author: I think that's what the second sentence (after the italics) aims to do. If it's OK, I will leave it as is for now?

> This page describes _automations_ in W&B. [Create an automation]({{< relref "create-automations/" >}}) to trigger workflow steps, such as automated model testing and deployment, based on an event in W&B, such as when an [artifact]({{< relref "/guides/core/artifacts" >}}) artifact version is created ({{< relref "/guides/core/registry/" >}}).

Author: ```suggestion
This page describes _automations_ in W&B. [Create an automation]({{< relref "create-automations/" >}}) to trigger workflow steps, such as automated model testing and deployment, based on an event in W&B, such as when an [artifact]({{< relref "/guides/core/artifacts" >}}) artifact version is created.
```

For example, an automation can post to a Slack channel when a new version is created, or can run a webhook to trigger automated testing when the `production` alias is added to an artifact.

## Overview
An automation can run when a specific [event]({{< relref "automation-events.md" >}}) occurs in a registry or project.

For an artifact in a [Registry]({{< relref "/guides/core/registry/">}}), you can configure an automation to run:
- When a new artifact version is linked to a collection. For example, trigger testing and validation workflows for new candidate models.

> - When an alias is added to an artifact version. For example, trigger a deployment workflow when the `deploy` alias is added to a model version.

Reviewer: Unsure how familiar users are with aliases compared to tags (and what they are used for). I'd omit this to prevent confusion. 

```suggestion
- When an alias is added to an artifact version. For example, trigger a deployment workflow when an alias is added to a model version.
```

- When an alias is added to an artifact version. For example, trigger a PII redaction workflow when the alias "redaction" is added to a dataset artifact.

For more details, refer to [Automation events and scopes]({{< relref "automation-events.md" >}}).

To [create an automation]({{< relref "create-automations/" >}}), you:

1. If required, configure [secrets]({{< relref "/guides/core/secrets.md" >}}) for sensitive strings the automation requires, such as access tokens, passwords, or sensitive configuration details. Secrets are defined in your **Team Settings**. Secrets are most commonly used in webhook automations to securely pass credentials or tokens to the webhook's external service without exposing it in plain text or hard-coding it in the webhook's payload.
1. Configure the webhook or Slack notification to authorize W&B to post to Slack or run the webhook on your behalf. A single automation action (webhook or Slack notification) can be used by multiple automations. These actions are defined in your **Team Settings**.
1. In the project or registry, create the automation:

>   1. Define the [event]({{< relref "#automation-events" >}}) to watch for, such as when a new artifact version is added. 

Reviewer: These nested numbers are not rendering correctly. They appear as a continuation of the list.

Author: Done (needed two more spaces each)

>   1. Define the [event]({{< relref "#automation-events" >}}) to watch for, such as when a new artifact version is added. 
>   1. Define the action to take when the event occurs (posting to a Slack channel or running a webhook). For a webhook, specify a secret to use for the access token and/or a secret to send with the payload, if required.

Author: ```suggestion
    1. Define the [event]({{< relref "#automation-events" >}}) to watch for, such as when a new artifact version is added. 
    1. Define the action to take when the event occurs (posting to a Slack channel or running a webhook). For a webhook, specify a secret to use for the access token and/or a secret to send with the payload, if required.
```

weight: 4
---

> This page describes _automations_ in W&B. [Create an automation]({{< relref "create-automations/" >}}) to trigger workflow steps, such as automated model testing and deployment, based on an event in W&B, such as when an [artifact]({{< relref "/guides/core/artifacts" >}}) is created or changed or a [registered model is changed]({{< relref "/guides/core/registry/" >}}).

Reviewer 2: ```suggestion
This page describes _automations_ in W&B. [Create an automation]({{< relref "create-automations/" >}}) to trigger workflow steps, such as automated model testing and deployment, based on an event in W&B, such as when an [artifact]({{< relref "/guides/core/artifacts" >}}) artifact version is created ({{< relref "/guides/core/registry/" >}}).
```

> For example, an automation can post to a Slack channel when a new artifact is created, or run a webhook to trigger automated testing when a model has the `test` label applied.

Reviewer 2: ```suggestion
For example, an automation can post to a Slack channel when a new version is created, or can run a webhook to trigger automated testing when the `production` alias is added to an artifact.
```

Author: I tweaked your suggestion slightly before applying because I think "run a webhook" makes more sense than "send a webhook".

## Overview
An automation can run when a specific [event]({{< relref "automation-events.md" >}}) occurs in a registry or project.

For an artifact in [Registry]({{< relref "/guides/core/registry/">}}), you can configure an automation to run:

> - When a new artifact is linked to a collection. For example, trigger testing and validation workflows for new models.

Reviewer 2: ```suggestion
- When a new artifact version is linked to a collection. For example, trigger testing and validation workflows for new candidate models.
```

> - When an alias is added to a new version of an artifact. For example, trigger a deployment workflow when the `deploy` alias is added to a model version.

Reviewer 2: ```suggestion
- When an alias is added to an artifact version. For example, trigger a deployment workflow when the `deploy` alias is added to a model version.
```

For an artifact in a project, you can configure an automation to run:

> - When a new version of an artifact is created. For example, start a training job when a new version of a dataset artifact is created.

Reviewer 2: ```suggestion
- When a new version is added to an artifact. For example, start a training job when a new version of a dataset artifact is added to a given collection.
```

> - When a new artifact is linked to a project or collection. For example, trigger testing and validation workflows for new models.

Reviewer 2: ```suggestion
```
this is the same event as above

> - When an alias is added to a new version of an artifact. For example, trigger a deployment workflow when the `deploy` alias is added to a model version.

Reviewer 2: ```suggestion
- When an alias is added to an artifact version. For example, trigger a PII redaction workflow when the alias "redaction" is added to a dataset artifact.
```

For more details, refer to [Automation events]({{< relref "automation-events.md" >}}).

To [create an automation]({{< relref "create-automations/" >}}), you:

1. If required, configure [secrets]({{< relref "/guides/core/secrets.md" >}}) for sensitive strings the automation requires, such as access tokens, passwords, or sensitive configuration details.

> 1. Authorize W&B to post to Slack or run the webhook on your behalf. This is required once per Slack channel or webhook.

Reviewer 2: ```suggestion
1. Authorize W&B to post to Slack or run the webhook on your behalf. This is required once per Slack channel or webhook.
```
can you explain more what this step means? Is this authorization the user has to do in W&B or in the webhook provider?

1. In the project or registry, create the automation:

>   1. Grant it access to any secrets you created for it.

Reviewer 2: ```suggestion
  1. Grant it access to any secrets you created for it.
```
Also not following what this step is - what does it mean to grant access to a secret?

>   1. Define the [event]({{< relref "#automation-events" >}}) to watch for, such as when a new artifact or version is created or linked.

Reviewer 2: ```suggestion
  1. Define the [event]({{< relref "#automation-events" >}}) to watch for, such as when a new artifact version is added. 
```
